### DISCUSSION TITLE
Execution Issues with FMU Files Generated by PythonFMU

### DISCUSSION THREAD

**Discussion by hsd0937** [NTNU-IHB/PythonFMU#201] [Q&A]
I have created the ManningFMU.fmu file using the pythonfmu library. However, when I import ManningFMU into MapleSim and attempt a simulation, an error occurs. FMU files created in Matlab work in MapleSim, but those created with pythonfmu do not seem to execute. It appears different from the FMU generated in C++. Could you please provide the code to set up pythonfmu files to run in Matlab or MapleSim?

![fmuerror](https://github.com/NTNU-IHB/PythonFMU/assets/36781692/a3b4c3bb-97f7-421e-b9df-b693cfb1bd2f)

Thank you.

**Comment by markaren** [comment]
It looks like there is no shared library in the FMU. Did you install from source without building the native library?

**Reply by hsd0937** [reply]
I did not build the native library. I installed pythonfmu in Python and created an FMU file. How do I install the native library?
Do you have a related sample code?

**Reply by markaren** [reply]
If you installed from pip/conda, there is no need to build the native library. Please check the content of the FMU.
Otherwise, build the CMake library located in pythonfmu/pythonfmu-export.

**Reply by hsd0937** [reply]
I have checked the CMake in pythonfmu/pythonfmu-export. Could you tell me how to build the CMake library?

**Reply by markaren** [reply]
Run the `build_XXX.sh` script. This assumes you have a working C++ toolchain.
